src/AutoMapper/XpressionMapper/FindMemberExpressionsVisitor.cs

```diff
@@ -44,19 +44,16 @@ public MemberExpression Result
 
         protected override Expression VisitMember(MemberExpression node)
         {
-            if (node.NodeType == ExpressionType.Constant)
-                return base.VisitMember(node);
-
             var parameterExpression = node.GetParameterExpression();
             var sType = parameterExpression?.Type;
             if (sType != null && _newParameter.Type == sType && node.IsMemberExpression())
             {
-                if (node.Expression.NodeType == ExpressionType.MemberAccess && (node.Type == typeof(string) 
-                                                                                    || node.Type.GetTypeInfo().IsValueType
-                                                                                    || (node.Type.GetTypeInfo().IsGenericType 
-                                                                                        && node.Type.GetGenericTypeDefinition() == typeof(Nullable<>)
-                                                                                        && Nullable.GetUnderlyingType(node.Type).GetTypeInfo().IsValueType)))
-                    _memberExpressions.Add((MemberExpression)node.Expression);
+                if (node.Expression.NodeType == ExpressionType.MemberAccess && (node.Type == typeof(string)
+                                                                                    || node.Type.GetTypeInfo().IsValueType))
+                    _memberExpressions.Add((MemberExpression)node.Expression);
+                else if (node.Expression.NodeType == ExpressionType.Parameter && (node.Type == typeof(string)
+                                                                                    || node.Type.GetTypeInfo().IsValueType))

```

Comment:
I think a little extension method encapsulating this test (string or value type) might make the code more readable. Of course I don't know much about expression mapping, just a thought :) A good name goes a long way.

Reply:
Makes sense  - done.